### ISSUE TITLE
focal loss for number of classes greater then 2

### ISSUE THREAD

**Issue by enrico310786** [AdeelH/pytorch-multi-class-focal-loss#10]
Hi,

I would like to know if this implementation is good even if the problem has a number of classes greater than two. Suppose I have four classes and the dataset is unbalanced with the following distribution [90%, 6%, 2%, 2%]. Should I then define an array of weights alpha for each class? For example alpha=torch.tensor([0.1, 0.94, 0.98, 0.98]) with the weights assigned to each class inversely proportional to their distribution? It's correct?

Thanks!

**Comment by AdeelH** [comment]
Yes, that's right. I recommend trying out some mock values so you get an intuition of how it differs from the unweighted loss.
